### PR TITLE
Implement Deno.kill for windows

### PR DESCRIPTION
**Pull request by disizali**
This PR is implementing `Deno.kill` for windows 
\+ re-enable ignored tests `[destroyed connection & serverTLS ]` for windows.

### CODE REVIEW

File: std/http/server_test.ts [hunk 1]
```diff
@@ -392,13 +396,13 @@ test({
 
 test({
   name: "serveTLS",
-  // FIXME(bartlomieju): hangs on windows, cause can't do `Deno.kill`
-  ignore: true,
+  ignore: false,
```
Comment (kerolloz) [inline std/http/server_test.ts:399]: ignore is false by default. I think it would be better if you remove `ignore: false`.

Comment (disizali) [reply std/http/server_test.ts]: right, thanks @kerolloz 👍

File: cli/signal.rs [hunk 2]
```diff
@@ -11,7 +21,33 @@ pub fn kill(pid: i32, signo: i32) -> Result<(), OpError> {
 }
 
 #[cfg(not(unix))]
-pub fn kill(_pid: i32, _signal: i32) -> Result<(), OpError> {
-  // TODO: implement this for windows
+pub fn kill(pid: i32, signal: i32) -> Result<(), OpError> {
+  unsafe {
+    match signal {
+      9 | 15 | 2 => {
```
Comment (piscisaureus) [inline cli/signal.rs:27]: Can you use symbolic names here (e.g. `SIGTERM`). Or at least add a comment describing what these signals mean?
Note that currently Deno.signals contains linux signals (that's a bug).

File: cli/signal.rs [hunk 3]
```diff
@@ -11,7 +21,33 @@ pub fn kill(pid: i32, signo: i32) -> Result<(), OpError> {
...
+        let handle = if pid == 0 {
+          GetCurrentProcess()
```
Comment (piscisaureus) [inline cli/signal.rs:29]: On other platforms, kill(0) does not kill the current process, but rather the current process group.
Closes equivalent on windows would be to terminate the current job object, but let's not implement that now.
Preferrably just error with "not supported".

File: cli/signal.rs [hunk 4]
```diff
@@ -11,7 +21,33 @@ pub fn kill(pid: i32, signo: i32) -> Result<(), OpError> {
...
+        } else {
+          OpenProcess(PROCESS_TERMINATE, 0, pid as DWORD)
```
Comment (piscisaureus) [inline cli/signal.rs:31]: Check that pid is positive before casting to DWORD. If pid < 0, fail with "not supported".

File: cli/signal.rs [hunk 5]
```diff
@@ -11,7 +21,33 @@ pub fn kill(pid: i32, signo: i32) -> Result<(), OpError> {
...
+        };
+
+        if handle.is_null() {
+          let m = format!("failed to open process : {}", pid);
+          return Err(OpError::other(m));
```
Comment (piscisaureus) [inline cli/signal.rs:36]: Preferrably report the OS error here (`std::io::Error::from_last_os_error()`).

File: cli/signal.rs [hunk 6]
```diff
@@ -11,7 +21,33 @@ pub fn kill(pid: i32, signo: i32) -> Result<(), OpError> {
...
+        }
+        if TerminateProcess(handle, 1) == 0 {
+          let m = format!("failed to terminate process : {}", pid);
+          return Err(OpError::other(m));
```
Comment (piscisaureus) [inline cli/signal.rs:40]: * On error, this leaks `handle`.
* Preferrably report the OS error here (`std::io::Error::from_last_os_error()`).

**Comment by piscisaureus** [comment]
Thanks, this is looking pretty good now.
Before we can land this we also need a test -- something simple where you start a process with Deno.run() and then kill it.

**Comment by piscisaureus** [comment]
LGTM, thanks @disizali!

**Pull request merged**